vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java

```diff
@@ -21,8 +21,7 @@ public StylingPage() {
         addClassNameFoo.setId("add-foo");
 
         Button setClassNameBar = new Button("Set class bar", e -> {
-            dialog.setClassName("foo bar");
-            dialog.getClassNames().set("foo", false);
+            dialog.setClassName("bar");
```

Comment:
Looks like the case in this IT isn't covered by the logic of `InternalOverlayClassListProxy` 😕 
However, this approach looks like an edge case to me, so we can probably accept it.

Reply:
This could be fixed by implementing `setClassName` to support multiple class names split by whitespace. Something like:

```java
public void setClassName(String className) {
    getClassNames().clear();
    if (className != null) {
        getClassNames().addAll(Arrays.asList(className.split(" ")));
    }
}
```

Reply:
Thanks, done.

Reply:
Makes sense to add the same change to `Dialog` also.

Reply:
Thanks, done.